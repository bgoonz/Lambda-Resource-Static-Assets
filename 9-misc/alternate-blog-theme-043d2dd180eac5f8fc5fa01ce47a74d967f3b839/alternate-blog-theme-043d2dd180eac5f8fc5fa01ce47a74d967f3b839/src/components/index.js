import Action from './Action';
import ActionIcon from './ActionIcon';
import Footer from './Footer';
import FormField from './FormField';
import Header from './Header';
import HeaderAlt from './HeaderAlt';
import Layout from './Layout';
import Icon from './Icon';

export {
    Action,
    ActionIcon,
    Footer,
    FormField,
    Header,
    HeaderAlt,
    Layout,
    Icon
};

export default {
    Action,
    ActionIcon,
    Footer,
    FormField,
    Header,
    HeaderAlt,
    Layout,
    Icon
};
