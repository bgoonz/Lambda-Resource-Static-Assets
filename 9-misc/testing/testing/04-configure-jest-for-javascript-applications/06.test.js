import React from 'react';
import {render} from 'react-testing-library';
import CalculatorDisplay from './src/calculator-display';
import {getFlyingSuperHeros} from './src/super-heros';

describe('super heros', () => {
  test('returns super heros that can fly', () => {
    const flyingHeros = getFlyingSuperHeros();

    expect(flyingHeros).toMatchSnapshot();
  });
});

describe('CalculatorDisplay', () => {
  test('mounts', () => {
    const {container} = render(<CalculatorDisplay value="0" />);

    // react-testing-library's render function returns its container property
    // wrapping your component in a div
    // Your component can be obtained at container.firstChild
    console.log(container.innerHTML);
    // console.log(container.firstChild);

    expect(container.firstChild).toMatchSnapshot();
  });
});
