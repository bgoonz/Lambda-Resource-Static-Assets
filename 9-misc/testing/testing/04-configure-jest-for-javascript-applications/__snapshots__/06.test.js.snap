// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorDisplay mounts 1`] = `
.emotion-0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 6em;
  line-height: 130px;
  position: relative;
}

<div
  class="emotion-0 emotion-1"
>
  <div
    class="autoScalingText"
    style="transform: scale(1,1);"
  >
    0
  </div>
</div>
`;

exports[`super heros returns super heros that can fly 1`] = `
Array [
  Object {
    "name": "Dynaguy",
    "powers": Array [
      "disintegration ray",
      "fly",
    ],
  },
  Object {
    "name": "Apogee",
    "powers": Array [
      "gravity control",
      "fly",
    ],
  },
]
`;
